core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableJoinRule.java

```diff
@@ -71,26 +71,25 @@
         EnumerableRules.LOGGER.debug(e.toString());
         return null;
       }
+    } else {
+      RelNode newRel;
+      try {
+        newRel = EnumerableHashJoin.create(
+            left,
+            right,
+            info.getEquiCondition(left, right, cluster.getRexBuilder()),
```

Comment:
This call is so redundant, when we create a join, it should have everything that is related with join info. Ideally, we should just call `join.getJoinInfo().getEquiCondition()` or just `join.getEquiCondition()`. But we can do it later.

Reply:
+1, will do it in another patch.